protostar-rust/src/cheatcodes_hint_processor.rs

```diff
@@ -105,6 +115,93 @@ struct ScarbStarknetContractArtifact {
     casm: Option<PathBuf>,
 }
 
+fn execute_syscall(
+    system: &ResOperand,
+    vm: &mut VirtualMachine,
+    blockifier_state: &mut CachedState<DictStateReader>,
+) -> Result<(), HintError> {
+    let (cell, offset) = extract_buffer(system);
+    let mut system_ptr = get_ptr(vm, cell, &offset)?;
+
+    let selector = felt_from_pointer(vm, &mut system_ptr)
+        .unwrap()
+        .to_bytes_be();
+
+    let gas_counter = usize_from_pointer(vm, &mut system_ptr).unwrap();
+    let contract_address = felt_from_pointer(vm, &mut system_ptr).unwrap();
+    let entry_point_selector = felt_from_pointer(vm, &mut system_ptr).unwrap();
+
+    let start = relocatable_from_pointer(vm, &mut system_ptr).unwrap();
+    let end = relocatable_from_pointer(vm, &mut system_ptr).unwrap();
+    let calldata = read_data_from_range(vm, start, end).unwrap();
+
+    assert_eq!(std::str::from_utf8(&selector).unwrap(), "CallContract");
```

Comment:
We cannot bubble standard errors from this metod because it returns `Result<(), HintError>`. We'd have to wrap them in HintError and panic data handling is being done in another PR.